internal/updater/extract.go

```diff
@@ -0,0 +1,77 @@
+package updater
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/bzip2"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/gopasspw/gopass/pkg/debug"
+	"github.com/pkg/errors"
+)
+
+func extractFile(buf []byte, filename, dest string) error {
+	var mode = os.FileMode(0755)
+
+	// if overwriting an existing binary retain it's mode flags
+	fi, err := os.Lstat(dest)
+	if err == nil {
+		mode = fi.Mode()
+	}
+
+	var rd io.Reader = bytes.NewReader(buf)
+	switch filepath.Ext(filename) {
+	case ".gz":
+		gzr, err := gzip.NewReader(rd)
+		if err != nil {
+			return err
+		}
+		rd = gzr
+	case ".bz2":
+		rd = bzip2.NewReader(rd)
+	case ".zip":
```

Comment:
So, the updater should exit early on Windows devices then.

Reply:
There's https://golang.org/pkg/archive/zip/ that could handle it.

Reply:
Yes, indeed. But I was too lazy do add windows support. I don't think that's much of a problem, but if this is working well on non-Windows OS we can still add Windows support later. Should be fairly easy to plug it in here.

Reply:
Sounds good.